crates/keys/src/algorithm.rs

```diff
@@ -0,0 +1,9 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum CryptoAlgorithm {
```

Comment:
For these specific things, it should be called "Curve" instead of "CryptoAlgorithm"

_Originally posted by @distractedm1nd in https://github.com/deltadevsde/prism/pull/187#discussion_r1892339475_

Reply:
I still disagree on this. What if we wanted to support keys/sigs without a curve?

    Are there signing keys that are not based on elliptic curves?
    
    RSA: Based on the factoring problem of large composite numbers. Still widely used.
    Hash-based signatures: SPHINCS+, XMSS, WOTS+. Very quantum-resistant but larger signatures.
    Lattice-based: Dilithium, Falcon. Also quantum-resistant, based on hard lattice problems.
    Multivariate: Rainbow (though broken now). Based on solving systems of multivariate polynomial equations.